samples/estimation/ShorRE.qs

```diff
@@ -0,0 +1,364 @@
+/// # Sample
+/// Resource Estimation for Shor's algorithm
+///
+/// # Description
+/// In this sample we concentrate on costing the `EstimateFrequency`
+/// operation, which is the core quantum operation in Shor's algorithm, and
+/// we omit the classical pre- and post-processing.
+namespace Shors {
```

Comment:
So, this is the quantum part of Shor? Would it make sense to try some new library arithmetic and see if estimates change?

Reply:
I think this one was tuned already to use `CCNOT` directly for AND in order to keep the counts consistent with our other samples. But definitely worth letting @msoeken take a look and refine in a later follow up too.

Reply:
Thanks for the feedback! I have incorporated that in both this and the simulator sample, after testing to confirm behavior is as expected.